app/src/main/java/com/training/simpleloginform/MainActivity.java

```diff
@@ -48,10 +71,15 @@ public void onClick(View view) {
 
                 usernameDetailsTextView.setText(username);
                 passwordDetailsTextView.setText(password);
+
+                names.add(username);
+                namesList.addAll(names);
+                names.clear();
```

Comment:
I don't think we have to clear the list. The list must be left untouched. If we want to avoid adding duplicate names we can loop through the names and check if the list contains the new name. If it does not, then we can add it. You can also use the contains() method. So something like if list.contains(someName) .... 

If you want to remove duplicates from the names list, you can do that by using your hash set, as you are currently doing but before the list is passed to the adapter.